### PR TITLE
#2133 Use commons version 1.1.0

### LINKED ISSUE #2133: Use commons version 1.1.0
## Background
A part of Enceladus has been extracted to _commons_ (https://github.com/AbsaOSS/commons/pull/108). Also some other functionalities used in Enceladus were in commons before.

## Feature
Use what is in the new release of _commons_(1.1.0) in Enceladus and remove what is no longer needed from the project.

### PR DESCRIPTION
**Pull request by jakipatryk**
- changed version of `absa.commons` library used to 1.1.0
- removed `Algorithms` and used implicit `SeqExtension` from `absa.commons` instead
- removed `ClassLoaderUtils` and used `ReflectionUtils.objectForName` from `absa.commons` instead
- removed `OptionImplicits` and used `OptionExtension` from `absa.commons` instead
- used `StringExtension` from `absa.commons` instead of `StringImplicits` (partially, as I missed newest functions [from 404 PR] in the commons PR)
- removed `LocalFsUtils`  and used `TempDirectory` from `absa.commons` instead of `getLocalTemporaryDirectory`, and `LocalFileSystemUtils` for others
- removed `DataFrameImplicitsSuite.scala` as this seems to be a leftover after `spark-commons` extraction or something

Closes #2133

### CODE REVIEW

**Review by dk1844 — DISMISSED** [review]
LGTM (just read the code)

File: spark-jobs/src/test/scala/za/co/absa/enceladus/common/plugin/ControlMetricsPluginSuite.scala [hunk 1]
```diff
@@ -90,7 +90,7 @@ class ControlMetricsPluginSuite extends AnyFunSuite {
       Map[String, String]("dummy.1" -> "za.co.absa.NoSuchClass")
         .asJava)
 
-    intercept[IllegalArgumentException] {
+    intercept[Exception] {
```
Comment (Zejnilovic) [inline spark-jobs/src/test/scala/za/co/absa/enceladus/common/plugin/ControlMetricsPluginSuite.scala:93]: Why so general? What is it supposed to throw?

Comment (benedeki) [inline spark-jobs/src/test/scala/za/co/absa/enceladus/common/plugin/ControlMetricsPluginSuite.scala:93]: Please revers to previous/higher class or explain why chosen such a generic class?
Same in `[PostProcessorPluginSuite.scala`.

**Review by benedeki — CHANGES_REQUESTED** [review]

File: spark-jobs/src/test/scala/za/co/absa/enceladus/common/plugin/ControlMetricsPluginSuite.scala (hunk 1 above)
Comment (jakipatryk) [reply spark-jobs/src/test/scala/za/co/absa/enceladus/common/plugin/ControlMetricsPluginSuite.scala]: Reverted back to IllegalArgumentException, catching exceptions thrown by the function from commons in PluginLoader and throwing IllegalArgumentException instead.

Comment (jakipatryk) [reply spark-jobs/src/test/scala/za/co/absa/enceladus/common/plugin/ControlMetricsPluginSuite.scala]: Reverted back to IllegalArgumentException, catching exceptions thrown by the function from commons in PluginLoader and throwing IllegalArgumentException instead.

**Review by benedeki — DISMISSED** [review]

**Review by dk1844 — APPROVED** [review]
LGTM (just read the code)

**Pull request merged**